source/common/config/watch_map.cc

```diff
@@ -62,6 +75,9 @@ void WatchMap::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>
     return;
   }
 
+  // Track any removals triggered by earlier watch updates.
+  deferred_removed_during_update_ = std::make_unique<absl::flat_hash_set<Watch*>>();
```

Comment:
Is the assumption here `deferred_removed_during_update_ == nullptr`, lets `ASSERT`?

Reply:
Nup, the goal is to reset the queue on reconnect.

Reply:
I'm bit confused, what happen to the existing set in there? Aren't we assume they are already removed?

Reply:
Right, I thought you were referring to another place we assign to a new unique_ptr. In this case, you are right, it should be null. Added the ASSERTs. Thanks.